src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,42 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exception.InvalidDataException;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int MIN_VALID_LENGTH = 6;
+    private static final int MINIMAL_AGE = 18;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        alreadyRegisteredValidator(user);
+        loginAndPasswordsValidator(user);
+        ageValidator(user);
+        return storageDao.add(user);
+    }
+
+    private void alreadyRegisteredValidator(User user) {
+        if (storageDao.get(user.getLogin()) != null) {
+            throw new InvalidDataException("User with current login is already registered");
+        }
+    }
+
+    private void loginAndPasswordsValidator(User user) {
```

Comment:
It's better to create two separate methods for validate login and password.
Also, it's better to throw exception with message if login or password are null and throw exception when length aren't ok